python/cudf_polars/cudf_polars/dsl/ir.py

```diff
@@ -2344,20 +2359,24 @@ def do_evaluate(
                 stream=stream,
             )
             if coalesce and how == "Full":
+                left_key_names = left_on.column_names
```

Comment:
Note: I'm not sure what happened but the column names for left and right came in unaligned, so that caused the dtypes to mismatch when calling `plc.replace.replace_nulls` -> Crash. This change makes sure the the column names and types are aligned.